crates/iroha_core/src/queue.rs

```diff
@@ -230,6 +233,9 @@ impl Queue {
                 err: Error::Full,
             }
         })?;
+        if let Err(err_hash) = self.tx_gossip.push(hash) {
+            warn!(tx=%err_hash, "Gossiper is lagging behind, not able to queue tx for gossiping");
+        }
```

Comment:
this transaction will never be gossiped and will enter the blockchain only when this peer becomes a leader. Although, I believe it's quite unlikely to happen

Reply:
Alternative is to have unlimited queue for gossiper, but if it can't keep up with the rate of transactions we would run out of memory.

Reply:
And broadcasting without batching can create network overhead

Reply:
I think it's fine, gossiping shouldn't be slower than block production